deeplake/core/dataset/deeplake_query_dataset.py

```diff
@@ -0,0 +1,375 @@
+                will be smaller. (default: ``False``)
+            return_index (bool): Showing wheter Loader needs to return the sample index during iteration.Defaults to True.
+            transform (Callable, optional): Callable object which is needed to be applyed on each sample on batch. Defaults to None.
+            num_workers (int): How many subprocesses to use for data
+                loading. ``0`` means that the data will be loaded in the main process.
+                (default: ``0``)
+            num_threads (int, optional): Number of threads that nedes to be spinned up during data loading. Defaults to None.
+                if it is none then we are detecting the hardware concurency count to set.
+                Note: We don't set any threading flags (eg. OMP_NUM_THREADS, MKL_NUM_THREADS, etc)
+                to get more performant Loader consider of not setting those flags which can affect on 3rd party libraries worflow performance
+            collate_fn (callable, optional): merges a list of samples to form a
+                mini-batch of Tensor(s).  Used when using batched loading from a
+                map-style dataset.
+            distributed (bool): Flag that is showing whether Loader needes to work in DDP or not. Defaults to ``False``
+            tensors (List[str], optinal): List of tensors thet are participating to in Loadeing process.
+                Defaults to ``None`` which means that Loader will fetch samples for all of the tensors
+            raw_tensors (List[str], optional): List of the tensors that needs to return raw data instead of decompression.
+                Defaults to ``None`` if raw_tensors is None then all the tensors will send decompression data
+                E.g raw_tensors['images'] then only the images tensor data will be sent as a row array
+            compressed_tensors (List[str], optional): Subset of raw tensors, these will be decompressed by python workers.
+            prefetch_factor (int): Number of samples loaded in advance by workers. Defaults to 10
+            upcast (bool): Flag that is showing wheter we need to upcast object if dtype is not supported this is needed only for
+                pytorch as it is not support all the dtypes. Defaults to True.
+            primary_tensor (Optional[str]): Name of primary tensor.
+            buffer_size (int): The size of the buffer used to shuffle the data in MBs. Defaults to 2048 MB. Increasing the buffer_size will increase the extent of shuffling.
+            persistent_workers (bool): If ``True``, the data loader will not shutdown the worker processes after a dataset has been consumed once. Defaults to ``False``.
+
+        Raises:
+            Exception: OSS dataloader is not supported on query dataset.
+        """
+        raise Exception(
+            "OSS dataloader is not supported for non-linear views. Use `view.dataloader().pytorch` instead."
+        )
+
+    def __getitem__(
```

Comment:
Can't this method be a wrapper over `Dataset.__getitem__`?

Reply:
If the only issue is the return class, you can change it to `self.__class__` in the base class.